### ISSUE TITLE
Single Fetch: Can't return empty Map or Set from loader (turbo-stream issue?)

### ISSUE THREAD

**Issue by caprica** [remix-run/remix#9495]
### Reproduction

https://stackblitz.com/edit/remix-run-remix-koagyf?file=app%2Froutes%2F_index.tsx

In this stackblitz, it will work initially because the Map/Set contains data, but comment out the indicated line of code that adds data to to the map or set and it will fail with an exception in the browser.

If you refresh the stackblitz with empty Map/Set it will initially render correctly, but then the page will clear, with console errors in the browser.

### System Info

```shell
System:
    OS: Linux 5.15 Linux Mint 21.3 (Virginia)
    CPU: (8) x64 Intel(R) Core(TM) i7-6700K CPU @ 4.00GHz
    Memory: 22.22 GB / 31.30 GB
    Container: Yes
    Shell: 5.1.16 - /bin/bash
  Binaries:
    Node: 18.20.2 - ~/.nvm/versions/node/v18.20.2/bin/node
    npm: 10.8.0 - ~/.nvm/versions/node/v18.20.2/bin/npm
  Browsers:
    Chrome: 123.0.6312.105
  npmPackages:
    @remix-run/dev: ^2.9.2 => 2.9.2 
    @remix-run/express: ^2.9.2 => 2.9.2 
    @remix-run/node: ^2.9.2 => 2.9.2 
    @remix-run/react: ^2.9.2 => 2.9.2 
    @remix-run/testing: 2.9.2 => 2.9.2 
    vite: ^5.2.11 => 5.2.11
```

### Used Package Manager

npm

### Expected Behavior

To successfully return empty Map and Set from loader and decode via useLoaderData.

### Actual Behavior

Unable to return empty Map or Set from a loader, would have to null it out instead.

Exception thrown in browser:

```
Error: Unable to decode turbo-stream response from URL: https://whatever
    at fetchAndDecode (https://whatever/node_modules/.vite/deps/@remix-run_react.js?v=6dff2301:2679:11)
```

Problem is likely in turbo-stream dependency used by Remix, I raised this issue a few weeks ago, trying to get more visibility here now: https://github.com/jacob-ebey/turbo-stream/issues/26

**Comment by caprica** [comment]
There is a change for this in turbo-stream release 2.1.0.

**Comment by brophdawg11** [comment]
This should be resolved by #9562 and available in the next release
